crates/cheatnet/tests/cheatcodes/spoof.rs

```diff
@@ -707,3 +733,232 @@ fn spoof_multiple() {
+    );
+}
+
+#[test]
+fn spoof_in_constructor_with_span() {
+    let mut cheatnet_state = CheatnetState::default();
+    let mut test_env = TestEnvironment::new(&mut cheatnet_state);
+
+    let contracts = get_contracts();
+
+    let class_hash = test_env.declare("ConstructorSpoofChecker", &contracts);
+    let precalculated_address = test_env
+        .runtime_state
+        .cheatnet_state
+        .precalculate_address(&class_hash, &[]);
+
+    let tx_info_mock = TxInfoMock {
+        transaction_hash: Some(Felt252::from(123)),
+        ..Default::default()
+    };
+
+    test_env.spoof(
+        CheatTarget::One(precalculated_address),
+        tx_info_mock,
+        CheatSpan::Number(2),
+    );
+
+    let contract_address = test_env.deploy_wrapper(&class_hash, &[]);
+    assert_eq!(precalculated_address, contract_address);
+
+    assert_success!(
+        test_env.call_contract(&contract_address, "get_transaction_hash", &[]),
+        vec![Felt252::from(123)]
+    );
+    assert_success!(
+        test_env.call_contract(&contract_address, "get_transaction_hash", &[]),
+        vec![TransactionHash::default().0.into_()]
+    );
+    assert_success!(
+        test_env.call_contract(&contract_address, "get_stored_tx_hash", &[]),
+        vec![Felt252::from(123)]
+    );
+}
+
+#[test]
+fn spoof_no_constructor_with_span() {
+    let mut cheatnet_state = CheatnetState::default();
+    let mut test_env = TestEnvironment::new(&mut cheatnet_state);
+
+    let contracts = get_contracts();
+
+    let class_hash = test_env.declare("SpoofChecker", &contracts);
+    let precalculated_address = test_env
+        .runtime_state
+        .cheatnet_state
+        .precalculate_address(&class_hash, &[]);
+
+    let tx_info_mock = TxInfoMock {
+        transaction_hash: Some(Felt252::from(123)),
+        ..Default::default()
+    };
+
+    test_env.spoof(
+        CheatTarget::One(precalculated_address),
+        tx_info_mock,
+        CheatSpan::Number(1),
+    );
+
+    let contract_address = test_env.deploy_wrapper(&class_hash, &[]);
+    assert_eq!(precalculated_address, contract_address);
+
+    assert_success!(
+        test_env.call_contract(&contract_address, "get_transaction_hash", &[]),
+        vec![Felt252::from(123)]
+    );
+    assert_success!(
+        test_env.call_contract(&contract_address, "get_transaction_hash", &[]),
+        vec![TransactionHash::default().0.into_()]
+    );
+}
+
+#[test]
+fn spoof_override_span() {
+    let mut cheatnet_state = CheatnetState::default();
+    let mut test_env = TestEnvironment::new(&mut cheatnet_state);
+
+    let contract_address = test_env.deploy("SpoofChecker", &[]);
+
+    let tx_info_before = test_env.get_tx_info(&contract_address);
+    let mut tx_info_mock = TxInfoMock {
+        transaction_hash: Some(Felt252::from(123)),
+        ..Default::default()
+    };
+    let expected_tx_info = TxInfo::apply_mock_fields(&tx_info_mock, &tx_info_before);
+
+    test_env.spoof(
```

Comment:
`start_spoof`?

Reply:
It's a valid use case to pass `CheatSpan::Indefinite` to `spoof` so I think it's fine to test it like that, but I'm glad you actually read any code during the review of these prs :D 🙏 ❤️